daemon/cmd/cells.go

```diff
@@ -220,8 +220,17 @@ var (
 		// Auth is responsible for authenticating a request if required by a policy.
 		auth.Cell,
 
-		// IPCache, policy.Repository and CachingIdentityAllocator.
-		cell.Provide(newPolicyTrifecta),
+		// Provides the IPCache
+		cell.Provide(newIPCache),
```

Comment:
Would be neater with single `cell.Provide` call (but either way is fine):
```
// (maybe comment why these grouped together and are in cmd package and what to do about it?)
cell.Provide(
   // Provides IPCache (IP to identity mappings)
  newIPCache, 
  // ...
  newPolicyRepo, 
  // ....
  newPolicyUpdater, 
  // ...
  newIdentityAllocator,
),
```

Reply:
Will try to move them into the a sub-package of the respective packages in a follow up PR. this way we might be able to break the circular build dependencies.

Reply:
@joamaki https://github.com/cilium/cilium/pull/34379 is a potential follow up